src/components/PageLayout/MenuMain/MenuMain.tsx

```diff
@@ -24,6 +24,25 @@ export const MenuMain: FC = () => {
   const [isVisible, setIsVisible] = useState(true)
   const toggleMenu = () => setIsVisible((currentIsVisible) => !currentIsVisible)
 
+  const fullWidthMenu = useMediaQuery<Theme>((theme) => theme.breakpoints.down('md'))
+  useEffect(() => {
+    fullWidthMenu && setIsVisible(false)
+  }, [fullWidthMenu])
+
+  // Inspired by https://nextjs.org/docs/pages/api-reference/functions/use-router#routerevents
+  const router = useRouter()
+  useEffect(() => {
+    const handleRouteChange = () => {
+      fullWidthMenu && setIsVisible(false)
+    }
+
+    router.events.on('routeChangeStart', handleRouteChange)
+
+    return () => {
+      router.events.off('routeChangeStart', handleRouteChange)
+    }
+  }, [router, fullWidthMenu])
```

Comment:
rozmyslam, ci je to potrebne... je to takto viac safe, ale realne myslim, ze MenuMain sa pri kazdom route changi vlastne remountuje, a tym padom by sa len vyhodnotil ten prvy effect a cajk 😄 anyway, ked to raz budeme prepisovat do app routeru a menu bude sucastou layoutu a bude zostavat mountnute, bude to mat urcite zmysel takto